usaspending_api/search/v2/views/spending_by_award.py

```diff
@@ -0,0 +1,259 @@
+                queryset = queryset.order_by(
+                    F("piid").desc(nulls_last=True), F("fain").desc(nulls_last=True), F("uri").desc(nulls_last=True)
+                ).values(*list(values))
+            else:
+                queryset = queryset.order_by(
+                    F("piid").asc(nulls_last=True), F("fain").asc(nulls_last=True), F("uri").asc(nulls_last=True)
+                ).values(*list(values))
+        elif self.order == "desc":
+            queryset = queryset.order_by(F(sort_filters[0]).desc(nulls_last=True)).values(*list(values))
+        else:
+            queryset = queryset.order_by(F(sort_filters[0]).asc(nulls_last=True)).values(*list(values))
+
+        limited_queryset = queryset[self.lower_bound : self.upper_bound]
+        has_next = len(limited_queryset) > self.limit
+
+        results = []
+        for award in limited_queryset[: self.limit]:
+            row = {"internal_id": award["award_id"]}
+
+            if award["type"] in loan_type_mapping:
+                for field in self.fields:
+                    row[field] = award.get(loan_award_mapping.get(field))
+            elif award["type"] in non_loan_assistance_type_mapping:
+                for field in self.fields:
+                    row[field] = award.get(non_loan_assistance_award_mapping.get(field))
+            elif award["type"] in idv_type_mapping:
+                for field in self.fields:
+                    row[field] = award.get(award_idv_mapping.get(field))
+            elif award["type"] in contract_type_mapping:
+                for field in self.fields:
+                    row[field] = award.get(award_contracts_mapping.get(field))
+
+            if "Award ID" in self.fields:
+                for id_type in ["piid", "fain", "uri"]:
+                    if award[id_type]:
+                        row["Award ID"] = award[id_type]
+                        break
+            results.append(row)
+
+        return {"limit": self.limit, "results": results, "page_metadata": {"page": self.page, "hasNext": has_next}}
+
+    def _handle_subawards(self, filters):
+        """ sub-award logic for SpendingByAward"""
+        # Test to ensure the award types are a proper subset of at least one group (contracts or grants)
+        if set(filters["award_type_codes"]) - set(procurement_type_mapping.keys()) and set(
+            filters["award_type_codes"]
+        ) - set(assistance_type_mapping.keys()):
+            error_msg = (
+                '{{"message": "\'award_type_codes\' array can only contain types codes of one group.",'
+                '"award_type_groups": {{'
+                '"procurement": {},'
+                '"assistance": {}}}}}'
+            ).format(json.dumps(procurement_type_mapping), json.dumps(assistance_type_mapping))
+            raise UnprocessableEntityException(json.loads(error_msg))
+
+        subawards_values = list(contract_subaward_mapping.keys()) + list(grant_subaward_mapping.keys())
```

Comment:
minor, but I feel like this should be `subawards_keys` or `subawards_fields`

Reply:
Previous name

Reply:
renamed to `field_external_name_list`